pycqed/measurement/calibration_toolbox.py

```diff
@@ -34,41 +34,52 @@ def mixer_carrier_cancellation_duplexer(**kw):
 
 def mixer_carrier_cancellation(SH, source, MC,
                                chI_par, chQ_par,
-                               frequency=None,
-                               voltage_grid=[
-                                   0.050, 0.020, 0.010, 0.005, 0.002],
-                               range_min=None,
-                               range_max=None):
+                               frequency: float=None,
+                               init_stepsize: float=0.1):
+    """
+    Varies the mixer offsets to minimize leakage at the carrier frequency.
+    this is a generic version.
+
+    Args:
+        SH     (instr) : Signal hound used to measure power
+        source (instr) : mw_source that provides the leakage tone
+        MC     (instr) :
+        chI_par (par)  :
+        chQ_par (par)  :
+        frequency (float) : the frequency in Hz at which to minimize leakage
+    """
 
-    ch_min = [0, 0]  # Initializing variables used later on
     source.on()
     if frequency is None:
-        frequency = source.get('frequency')
+        frequency = source.frequency()
     else:
-        source.set('frequency', frequency)
+        source.frequency(frequency)
 
-    for voltage_span in voltage_grid:
-        # Channel 0
-        for i, ch_par in enumerate([chI_par, chQ_par]):
-            MC.set_sweep_function(ch_par)
-            MC.set_detector_function(
-                det.Signal_Hound_fixed_frequency(signal_hound=SH, frequency=frequency))
-
-            swp_start = ch_min[i] + voltage_span
-            swp_end = ch_min[i] - voltage_span
-            swp_pts = np.linspace(swp_start, swp_end,  11)
-            print(swp_pts)
-            MC.set_sweep_points(swp_pts)
-            MC.run(name='Mixer_cal_{}'.format(ch_par.name),
-                   sweep_delay=.1)
-            Mixer_Calibration_Analysis = ma.Mixer_Calibration_Analysis(
-                label='Mixer_cal')
-            ch_min[i] = Mixer_Calibration_Analysis.fit_results[0]
-            ch_par(ch_min[i])
-
-    chI_par(ch_min[0])
-    chQ_par(ch_min[1])
-    return ch_min
+    '''
+    Make coarse sweeps to approximate the minimum
+    '''
+
+    detector = det.Signal_Hound_fixed_frequency(
+        SH, frequency=(source.frequency()),
```

Comment:
please set the reference level to -40 dB to improve noise floor of the signal hound with:

SH.ref_level(-40)